biz.aQute.bndlib/src/aQute/bnd/build/Workspace.java

```diff
@@ -725,6 +725,9 @@ public AtomicBoolean getOffline() {
 
 	public Workspace setOffline(boolean on) {
 		offline.set(on);
+		for (HttpClient client : getPlugins(HttpClient.class)) {
+			client.setOffline(on);
```

Comment:
I changes the code to use an AtomicBoolen to hold the offline state. You need to send that object into the HttpClient (not its current value) and then have the HttpClient call get on the object. Setting the AtomicBoolean into the HttpClient can be done when constructed and not at every all to  Workspace.setOffline.